### PR TITLE
Update z3 on docker files

### PR DESCRIPTION
**Pull request by leonardoalt**
Depends on https://github.com/ethereum/solidity/pull/9797

### CODE REVIEW

File: scripts/docker/buildpack-deps/Dockerfile.ubuntu1604.clang.ossfuzz
Comment (cameel) [inline scripts/docker/buildpack-deps/Dockerfile.ubuntu1604.clang.ossfuzz:59]: I wonder why it didn't fail if it was still at 4.8.7 after the tests were updated for 4.8.8.

Comment (leonardoalt) [reply scripts/docker/buildpack-deps/Dockerfile.ubuntu1604.clang.ossfuzz:59]: Because that job doesn't run `soltest`, just `solfuzzer`, so the error expectation is not checked.

Comment (ekpyron) [reply scripts/docker/buildpack-deps/Dockerfile.ubuntu1604.clang.ossfuzz:59]: I also wonder whether the google image is actually still based on ubuntu 16.04 at all, or whether they moved on by now :-)... difficult to tell just from the file here :-). Not that it matters too much anyways, it's just a name...

**Comment by ekpyron** [comment]
I intentionally retargetted this to ``develop`` (and apparently that *didn't* retrigger the docker image builds - which is good).

Since ``t_archlinux_soltest`` is not "required" we should be able to merge this once I approve, which I'll do once the tests in https://github.com/ethereum/solidity/pull/9797 pass.

**Review by ekpyron — APPROVED** [review]

**Pull request merged**
